src/mmap.rs

```diff
@@ -427,9 +447,30 @@ impl GuestMemoryMmap {
             }
         }
 
-        Ok(Self {
-            regions: Arc::new(regions),
-        })
+        Ok(Self { regions })
+    }
+
+    /// Add a new `GuestRegionMmap` to a vector of regions.
```

Comment:
nit: You're leaking the implementation details in the comment.  You can simply say "Add a new region to the set of existing regions."

Reply:
done!